commands/core/init.drush.inc

```diff
@@ -0,0 +1,115 @@
+  );
+  return $items;
+}
+
+/**
+ * Initialize local Drush configuration
+ */
+function drush_init_core_init() {
+  $home = drush_server_home();
+  $drush_config_dir = $home . "/.drush";
+  $drush_config_file = $drush_config_dir . "/drushrc.php";
+  $drush_bashrc = $drush_config_dir . "/drush.bashrc";
+  $examples_dir = DRUSH_BASE_PATH . "/examples";
+  $example_configuration = $examples_dir . "/example.drushrc.php";
+  $example_bashrc = $examples_dir . "/example.bashrc";
+  $bashrc_additions = array();
+
+  // Create a ~/.drush directory if it does not yet exist
+  if (!is_dir($drush_config_dir)) {
+    drush_mkdir($drush_config_dir);
+  }
+
+  // If there is no ~/.drush/drushrc.php, then copy the
+  // example Drush configuration file here
+  if (!is_file($drush_config_file)) {
+    @copy($example_configuration, $drush_config_file);
+    drush_log(dt("Copied example Drush configuration file to !path", array('!path' => $drush_config_file)), 'ok');
+  }
+
+  // If there is no ~/.drush/drush.bashrc file, then copy
+  // the example bashrc file here
+  if (!is_file($drush_bashrc)) {
+    @copy($example_bashrc, $drush_bashrc);
+    $bashrc_additions = array("%$drush_bashrc%" => "# Include Drush bash customizations\n. $drush_bashrc\n\n");
+    drush_log(dt("Copied example Drush bash configuration file to !path", array('!path' => $drush_bashrc)), 'ok');
+  }
+
+  // Decide whether we want to add our Bash commands to
+  // ~/.bashrc or ~/.bash_profile
+  $bashrc = drush_init_find_bashrc($home);
+
+  // If drush is not in the $PATH, then figure out which
+  // path to add so that Drush can be found globally.
+  if (!drush_which("drush")) {
+    $drush_path = drush_find_path_to_drush();
+    $drush_path = preg_replace("%^$home/%", '$HOME/', $drush_path);
+
+    $bashrc_additions = array("%$drush_path%" => "# Path to Drush, added by 'drush init'\n\$PATH=\$PATH:$drush_path\n\n");
+  }
+
+  // Modify the users bashrc file, adding our customizations.
+  $bashrc_contents = "";
+  if (file_exists($bashrc)) {
+    $bashrc_contents = file_get_contents($bashrc);
+  }
+  $new_bashrc_contents = $bashrc_contents;
+  foreach ($bashrc_additions as $pattern => $addition) {
+    // Only put in the addition if the pattern does not already
+    // exist in the bashrc file.
+    if (!preg_match($pattern, $new_bashrc_contents)) {
+      $new_bashrc_contents = $addition . $new_bashrc_contents;
+    }
+  }
+  if ($new_bashrc_contents != $bashrc_contents) {
+    file_put_contents($bashrc, $new_bashrc_contents);
+    drush_log(dt("Updated bash configuration file !path", array('!path' => $bashrc)), 'ok');
+  }
+}
+
+/**
+ * Determine which .bashrc file is best to use on this platform.
+ */
+function drush_init_find_bashrc($home) {
+  return $home . "/.bashrc";
+}
+
+/**
+ * Determine where Drush is located, so that we can add
+ * that location to the $PATH
+ */
+function drush_find_path_to_drush($home) {
+  // First test: is Drush inside a vendor directory?
```

Comment:
dirname(DRUSH_COMMAND) == DRUSH_BASE_PATH; I could use either, as they both indicate the same thing. Switching to dirname(DRUSH_COMMAND) would be fine if you think that is a more documented / more official constant than DRUSH_BASE_PATH.

In the use-case where we are looking for the location of the Drush executable, the Drush Finder and Launcher that were used in this execution of Drush will always be in the same directory as drush.php. If a different Drush Finder was in use, then presumably it is already on the $PATH.

One use case we might want to consider improving here is where the user creates a new Drupal 8 site via composer create-project, and then runs `./vendor/bin/drush init`. This is a fairly compelling use-case, but in the current implementation, you end up with the current site-local Drush in your $PATH, which might not be what you want. It might seem odd for Drush to install and/or copy itself to a "global" location, but perhaps a `composer global install drush` is what is needed here.  Was uncertain about that, so I just let it select the site-local Drush for now.